lib/utils/getServerlessConfigFile.js

```diff
@@ -46,28 +50,34 @@ const getServerlessConfigFilePath = serverless => {
   );
 };
 
-const handleJsConfigFile = jsConfigFile =>
+const handleJsOrTsConfigFile = (configFile, isTs) =>
   BbPromise.try(() => {
-    // use require to load serverless.js
+    if (isTs) {
+      // eslint-disable-next-line global-require
+      require('ts-node').register();
```

Comment:
I would prefer not add it as dependency. I think TS users should just ensure that both `typescript` and `ts-node` are installed in their setup.

Also we need to ensure it'll work with _standalone_ build, which makes it a bit more tricky (as it doesn't have natural access to _service_ or _global_ npm dependencies)

I imagine reliable logic of acquisition of `ts-node` as follows:

```javascript
const resolveModulePath = require('ncjsm/resolve');
const spawn = require('child-process-ext/spawn')
...

const tsNode = () => {
	// 1. Resolve as serverless peer dependency
	return resolveModulePath(__dirname, "ts-node")
		.catch((error) => {
			if (error.code !== "MODULE_NOT_FOUND") throw error;
			// 2. Resolve as service dependency
			return resolveModulePath(path.dirname(configFile), "ts-node");
		})
		.catch((error) => {
			if (error.code !== "MODULE_NOT_FOUND") throw error;
			// 2. Resolve as global installation
			return spawn("npm", ["root", "-g"]).then(
				({ stdoutBuffer }) => {
					return resolveModulePath("/", `${String(stdoutBuffer).trim()}/ts-node`).catch(
						(error) => {
							if (error.code !== "MODULE_NOT_FOUND") throw error;
							return null;
						}
					);
				},
				(error) => null
			);
		})
		.then((tsNodePath) => {
			if (!tsNodePath) {
				throw new ServerlessError(
					"Ensure 'ts-node' dependency for working with TypeScript configuration files"
				);
			}
			return require(tsNodePath);
		});
};
```

Reply:
I updated this to remove ts-node and resolve it here - some weird stuff I had to do to get it to resolve symlinks (because it returns an object instead of filepath), but works for all 3 cases.  Let me know if you like this approach, or if you want to include ts-node as a dep.